libra-swarm/src/swarm.rs

```diff
@@ -287,28 +293,27 @@ impl LibraSwarm {
     }
 
     pub fn configure_swarm(
+        template: Option<NodeConfig>,
         num_nodes: usize,
         role: RoleType,
         config_dir: Option<String>,
-        template_path: Option<String>,
         upstream_config_dir: Option<String>,
     ) -> Result<LibraSwarm> {
         let swarm_config_dir = Self::setup_config_dir(&config_dir);
         info!("logs at {:?}", swarm_config_dir);
-        let mut template = if let Some(template_path) = template_path {
-            NodeConfig::load(workspace_builder::workspace_root().join(template_path))
-                .unwrap_or_default()
-        } else {
-            let mut template = NodeConfig::default();
-            template.vm_config.publishing_options = VMPublishingOption::Open;
+
+        let mut node_config = if let Some(template) = template {
             template
+        } else {
+            NodeConfig::default()
         };
-        template.base.role = role;
+        node_config.vm_config.publishing_options = VMPublishingOption::Open;
+        node_config.base.role = role;
```

Comment:
we should actually move this to the only test that needs it....

Reply:
related to previous discussion, but I'd at least move it out of swarm.rs. Maybe to configuring defaults in smoke test itself